### DISCUSSION TITLE
FulfillmentInbound

### DISCUSSION THREAD

**Discussion by ghost** [abuzuhri/Amazon-SP-API-CSharp#105] [Show and tell]
Hi. First of all, this library is good.
I wanted to ask you, if you could show me how to upload TrackingIds for FulfillmentInboundShippings. I have a problem with this part, also with NonPartneredSmallParcelPackageInput.

Thank you !

**Comment by abuzuhri** [comment]
Hi @qemajlosmanihub 

Sorry for late reply , can you send all error details

**Reply by ghost** [reply]
Hello. Thank you for your response. I found some bugs in library that I fixed. Tomorrow I can send you all the changes and fixed I did to put this to work. Thanks.

**Reply by abuzuhri** [reply]
You are welcome , send I will change it

**Reply by abuzuhri** [reply]
also for easy for you in future you can fork the library and fix the bug in your branch and then submit Pull Request to me its very easy in Github

**Reply by abuzuhri** [reply]
@qemajlosmanihub 

Any Update on the bugs you said

**Reply by ghost** [reply]
Hi @abuzuhri. I will send you my changes today later. I am at work now and I can't do it. Sorry

**Comment by ghost** [comment]
1. AmazonSpApiSDK.Models.FulfillmentInbound.UnitOfMeasurement - In this class were only centimeters and inches. For some reason, Amazon had another member CM. I had to insert new Enum member here: [EnumMember(Value = "CM")]

2. AmazonSpApiSDK.Models.FulfillmentInbound.UnitOfWeight - For some reason this did'nt worked and I had to do this:
[JsonConverter(typeof(StringEnumConverter))] -- when i added this, then it worked
    public enum UnitOfWeight
    {

        /// <summary>
        /// Enum Pounds for value: pounds
        /// </summary>
        [EnumMember(Value = "pounds")]
        Pounds = 1,

        /// <summary>
        /// Enum Kilograms for value: kilograms
        /// </summary>
        [EnumMember(Value = "kilograms")]
        Kilograms = 2
    }

3. I also had problems with Class Members data types, for example, we here in Germany use this format, example: 1.20, 45.2 instead of 1,20 or 45,2. I faced problems when property datatype was set as BigDecimalType. I had to change this to Double or Decimal and then it worked good.
4. More or less, everywhere where it was BigDecimalType , I had to change it to Decimal or Double.
5. AmazonSpApiSDK.Models.FulfillmentInbound.TrackingId
[JsonConstructorAttribute]
public TrackingId(){}
Also this did not worked for me. I had to use property string and find a way to workaround.

I faced also some other models and classes missed some things, but honestly I don't know where and what I've changed.

**Reply by abuzuhri** [reply]
Do you Add this changed to your folk ?
https://github.com/qemajlosmanihub/Amazon-SP-API-CSharp/commits/main

**Reply by abuzuhri** [reply]
> I faced also some other models and classes missed some things, but honestly I don't know where and what I've changed.

Can you share the source code for the project i will check diff

**Reply by ghost** [reply]
No I did not added those changes to my Fork. Tomorrow, maybe I can send you the source code. Thanks

**Reply by abuzuhri** [reply]
> No I did not added those changes to my Fork. Tomorrow, maybe I can send you the source code. Thanks

Any update here ?
